### ISSUE TITLE
'ghci' is not recognized as an internal or external command and `Couldn't start client Haskell HIE (haskell)`

### ISSUE THREAD

**Issue by giuliohome** [haskell/haskell-ide-engine#878]
I'm running hui on windows with stack.
If I run the hui.exe

```
C:\Users\MyHome\AppData\Roaming\local\bin>hie.exe
2018-10-15 00:52:11.4374875 [ThreadId 3] - Using plain GHC version
"ghc" non è riconosciuto come comando interno o esterno,
 un programma eseguibile o un file batch.
hie.exe: readCreateProcess: ghc --version (exit 1): failed
```
I have to do `stack exec hie.exe`

```
C:\Users\MyHome>stack exec hie
2018-10-15 01:10:31.7484217 [ThreadId 3] - Using plain GHC version
2018-10-15 01:10:31.7924242 [ThreadId 3] - Mismatching GHC versions: Project is 8.2.2, HIE is 8.4.3
2018-10-15 01:10:31.7974245 [ThreadId 3] - Run entered for HIE(hie.EXE) Version 0.2.2.0, Git revision d9dc64ebd11e5a1f177476674a0939a634b25046 (2020 commits) x86_64 ghc-8
2018-10-15 01:10:31.806425 [ThreadId 3] - Current directory:C:\Users\MyHome
```

or I have to move to the source dir and start hie from there

```
C:\haskell\ide\haskell-ide-engine>hie
2018-10-15 01:09:36.8062792 [ThreadId 3] - Using stack GHC version
2018-10-15 01:09:37.048293 [ThreadId 3] - Run entered for HIE(hie) Version 0.2.2.0, Git revision d9dc64ebd11e5a1f177476674a0939a634b25046 (2020 commits) x86_64 ghc-8.4.3
2018-10-15 01:09:37.0572935 [ThreadId 3] - Current directory:C:\haskell\ide\haskell-ide-engine
```

In any case I'm getting `Couldn't start client Haskell HIE (haskell)` from Visual Studio Code

**Comment by giuliohome** [comment]
It seems to be resolved by starting `hie-wrappe`r, i.e.  by setting path 
`~/AppData/Roaming/local/bin/hie-wrapper`
But then I get the following error:
`The Haskell HIE (haskell) server crashed 5 times in the last 3 minutes. The server will not be restarted.`

**Comment by giuliohome** [comment]
I guess you should do maybe something like

```
pr <- fromMaybe (error "No project root found") <$> getProjectRootCurrent stackYmlExists <- doesFileExist (pr </> "stack.yaml")
```

instead of the simple `isStackProject <- doesFileExist "stack.yaml"`

but more generally I wonder whether this has been tested with stack ghc version on windows and with vs code :-/

**Comment by lorenzo** [comment]
Does the suggestion above solve your problem?

**Comment by giuliohome** [comment]
Let me try before if I can solve the problem without a commit: I wish to try to enter `stack exec hie-wrapper` in some vs studio settings parameter if it makes sense. Otherwise, agreed, I will follow your suggestion and compile the above change. Either way I'll let you know later today or tomorrow. Thanks for the feedback.

**Comment by giuliohome** [comment]
@lorenzo yep, forcing [this if](https://github.com/haskell/haskell-ide-engine/blob/master/src/Haskell/Ide/Engine/Plugin/Base.hs#L111) to `True`, rebuilding and deploying the `hie-wrapper` from `dist` to `local` `bin` does the trick! 
I've solved the problem myself now in this quick and dirty way.

**Comment by bubba** [comment]
@giuliohome is `hie` being run in the directory containing the `stack.yaml`? It should automatically pick that up, but sometimes a rogue `.ghc-environment-*` or `dist`/`dist-newstyle` can throw it off

**Comment by giuliohome** [comment]
@bubba hie is being run by vscode hie extension. My environment has no problem outside hie itself. 
Sorry but your answer seems fake for 2 reasons:
1. hie doesn't really read the content of stack.yaml but it has only to determine if it has to use stack or plain ghc. I have followed the instructions to install and configure hie for vscode and there's something clearly wrong and not working when the build and execution are done by stack
2. I have already solved this issue for myself an shared my workaround. So it's no longer an issue for me.  It's up to the community and not to me to decide to close or ignore this or to fix the code and the docs. To be blunt I'm also quite disappointed by the amount of time and space wasted for this service (30 minutes to build two lines of code, 20 giga of disk space lost, a lot of constraints limiting the actual usage in vscode) and so I'm not willing to subscribe this thread and follow up again. Sorry
